src/app/handlers/command.handler.ts

```diff
@@ -143,13 +182,26 @@ export class CommandHandler extends Handler {
     const pEvent: types.IPluginEvent = {
       status: 'starting',
       pluginName: plugin.name,
-      args: command.args,
-      user: message.author.tag,
+      args: [],
+      user,
     };
 
     try {
       this.container.loggerService.info(JSON.stringify(pEvent));
-      await plugin.execute(message, command.args);
+
+      // If it's a regular command, use the regular executor.
+      if (commandData.type === 'message') {
+        await plugin.execute(commandData.message, commandData.command.args);
+      }
+
+      // If it's a slash command, use the slash runner.
+
+      // This should never happen.
+      if (!types.isSlashCommand(plugin)) {
+        throw new Error('Cannot invoke slash command on a non-slash-command plugin.');
+      }
+
+      await (plugin as unknown as ISlashPlugin).run(message as CommandInteraction);
```

Comment:
?

Reply:
There's some trickiness going on with how exactly we want to tag slash command plugins vs classic plugins. I can rework some of this if you don't like how it is currently.